v4/newrelic/transaction.go

```diff
@@ -63,11 +80,19 @@ func (txn *Transaction) isEnded() bool {
 }
 
 // Ignore prevents this transaction's data from being recorded.
-func (txn *Transaction) Ignore() {}
+func (txn *Transaction) Ignore() {
+	if txn != nil && txn.thread != nil {
```

Comment:
I went back and forth between having these two checks in a single method (`txn.canLog()`) and I can go either way on it. We check it enough that it makes sense to have it in its own method to me, but at some point I felt differently and I haven't changed it back. 🤷‍♀️

Reply:
You have a `thd.canLog()`, but not a `txn.canLog()`. I think I'm slightly inclined to have `canLog()`'s as they are DRY'er. I can go either way, but I do think we should be consistent.

Reply:
Cool, I agree - I'll change it back to better reuse the code.